ut_core.py

```diff
@@ -0,0 +1,184 @@
+
+import sys
+import yaml
+import re
+
+from framework.core.logModule import logModule
+
+class UTCoreMenuNavigator:
+    """
+    Navigates through the UTcore menu system, trigger the execution of test cases, and collect results.
+    """
+
+    def __init__(self, menu_config_path, console):
+        """
+        Initializes the UTCoreMenuNavigator object with a menu configuration file and an optional test profile.
+
+        Args:
+            menu_config_path (str): The file path to the menu configuration YAML file.
+            console: The console session object to communicate with the UTcore system.
+            test_profile_url (str, optional): The URL to a test profile YAML file. Defaults to None.
+        """
+        self.session = console
+        self.menu_config = self.load_yaml(menu_config_path)
+        self.shell = self.session.open_interactive_shell()
+        self.log = logModule("UTCoreMenuNavigator")
+
+    
+    def find_key(self, d, target_key):
+        """
+        Recursively searches for a target key in a nested dictionary and returns the corresponding value.
+
+        Args:
+            d (dict): The dictionary to search in.
+            target_key (str): The key to search for.
+
+        Returns:
+            The value associated with the target key if found, else None.
+        """
+        if target_key in d:
+            return d[target_key]
+        
+        for key, value in d.items():
+            if isinstance(value, dict):
+                result = self.find_key(value, target_key)
+                if result is not None:
+                    return result
+        return None
+
+
+    def load_yaml(self, path):
+        """
+        Loads and parses a YAML file from the specified path.
+
+        Args:
+            path (str): The file path to the YAML file.
+
+        Returns:
+            dict: The parsed YAML content.
+        """
+        with open(path, 'r') as file:
+            return yaml.safe_load(file)
+        
+
+    def run_commands(self, run_script_path, group_name, test_name):
```

Comment:
this function needs separation, done in the next ticket.

run & selecting menu items are two separate things.

select_menu_item() - get's called by the upper API's multiple times and the current binary is still alive.